init.lua

```diff
@@ -77,21 +77,105 @@ local function get_v(v)
 	return math.sqrt(v.x ^ 2 + v.z ^ 2)
 end
 
+local function ensure_passengers_exists(entity)
+	if entity.passengers ~= nil then
+		return
+	end
+	entity.passengers = {}
+	if entity.passenger ~= nil or
+		entity.passenger_attach_at ~= nil or
+		entity.passenger_eye_offset ~= nil then
+		table.insert(entity.passengers,{
+			player=entity.passenger,
+			attach_at=entity.passenger_attach_at,
+			eye_offset=entity.passenger_eye_offset
+		})
+	else
+		return
+	end
+	if entity.passenger2 ~= nil or
+		entity.passenger2_attach_at ~= nil or
+		entity.passenger2_eye_offset ~= nil then
+		table.insert(entity.passengers,{
+			player=entity.passenger2,
+			attach_at=entity.passenger2_attach_at,
+			eye_offset=entity.passenger2_eye_offset
+		})
+	else
+		return
+	end
+	if entity.passenger3 ~= nil or
+		entity.passenger3_attach_at ~= nil or
+		entity.passenger3_eye_offset ~= nil then
+		table.insert(entity.passengers,{
+			player=entity.passenger3,
+			attach_at=entity.passenger3_attach_at,
+			eye_offset=entity.passenger3_eye_offset
+		})
+	end
+end
+-- Copies the specified passenger to the older api. Note that this is one-directional.
+-- If something changed in the old api before this is called it is lost.
+-- In code you control be sure to always use the newer API and to call this function on every change.
+-- If you would like to improove preformance (memory & CPU) by not updating the old API, set
+--  entity.dont_use_old_passenger_api to true. This will return from the funciton instead of doing anything.
+local function copy_a_passenger_to_old_api(entity,index,player,attach,eye)
```

Comment:
For the function's name, I'd like more `old_copy_passenger`. What do you think?

Reply:
Sure. These were placeholder names sorta anyway.